akka-stream-tck/src/test/scala/akka/stream/tck/ActorSystemLifecycle.scala

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.tck
+
+import java.util.concurrent.TimeoutException
+import scala.concurrent.duration._
+import akka.actor.ActorSystem
+import akka.actor.ActorSystemImpl
+import org.testng.annotations.AfterClass
+import akka.stream.testkit.AkkaSpec
+import akka.event.Logging
+import akka.testkit.TestEvent
+import akka.testkit.EventFilter
+
+trait ActorSystemLifecycle {
+
+  /**
+   * Implement this in the concrete test by:
+   * {{
+   * override lazy val system = createActorSystem()
+   * }}
+   *
+   * TestNG creates 2 instances of the test class and one of
+   * them is not used and @AfterClass is not invoked on one of
+   * the instances, i.e. the system is not shutdown.
```

Comment:
Is there a `@BeforeClass` annotation? maybe the actor system should be created there and store in a var. The `AkkaJUnitActorSystemResource` works in a similar way.